pkg/kubectl/cmd/apply_test.go

```diff
@@ -279,6 +296,85 @@ func TestApplyObject(t *testing.T) {
 	}
 }
 
+// TestApplyTPR uses apply commad with ThirdPartyResources.
+func TestApplyTPR(t *testing.T) {
+	initTestErrorHandler(t)
+	nameTPR := "test-tpr.unit-test.test.com"
+	pathTPR := "/thirdpartyresources/" + nameTPR
+	currentTPR := readTPRFromFile(t, filenameTPR)
+
+	verifiedPatch := false
+
+	f, tf, _, ns := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.JSONEncoder = testapi.Extensions.Codec()
+	tf.UnstructuredClient = &fake.RESTClient{
+		ContentConfig: restclient.ContentConfig{
+			ContentType: runtime.ContentTypeJSON,
+			GroupVersion: &schema.GroupVersion{
+				Group:   extensions.GroupName,
+				Version: "v1beta1",
+			},
+			NegotiatedSerializer: ns,
+		},
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == pathTPR && m == "GET":
+				bodyTPR := ioutil.NopCloser(bytes.NewReader(currentTPR))
```

Comment:
`currentTPR ` you return doesn't have the `last-applied-config`. Apply need to that annotation. Please check other test cases.

I suggest you have at least 2 approaches:
1) use json to manipulate that field.
2) create a TPR obj with `last-applied-config` in the test yaml file.